internal/codependencies/README.md

```diff
@@ -0,0 +1,12 @@
+This package allows us to depend on other modules, to _force_ these other
+modules to upgrade to some minimum version. This allows us to express "co
+dependency" requirements in cases where this module doesn't strictly speaking
+_depend_ on another module, but conflicts with some version of that module.
+
+In practice, this means:
+
+1. Packages imported here _will not_ end up in the final binary as nothing
+   imports this package.
+2. Modules containing these packages will, unfortunately, be downloaded as
+   "dependencies" of this package.
+3. Anyone using this module will be forced to upgrade all co-dependencies.
```

Comment:
It's not immediately clear from this README.md that these imports are actually deprecated and absorbed into go-multiaddr. Can we make this clear in the README.md?

Anecdote: I had forgotten we had (finally) consolidated these libs, and was about to review this with "if we're needing to do this, it's a smell that those libraries should actually live here."

Reply:
@raulk filter is not yet moved in here, only net is